qiskit_machine_learning/neural_networks/estimator_qnn.py

```diff
@@ -98,13 +118,20 @@ def __init__(
             estimator: The estimator used to compute neural network's results.
                 If ``None``, a default instance of the reference estimator,
                 :class:`~qiskit.primitives.Estimator`, will be used.
-            circuit: The quantum circuit to represent the neural network.
+            circuit: The quantum circuit to represent the neural network. If a
+                :class:`~qiskit_machine_learning.circuit.library.QNNCircuit` is passed,
+                input_params and weight_params are omitted. In this case these two properties are
+                held in the :class:`~qiskit_machine_learning.circuit.library.QNNCircuit`.
             observables: The observables for outputs of the neural network. If ``None``,
                 use the default :math:`Z^{\otimes num\_qubits}` observable.
             input_params: The parameters that correspond to the input data of the network.
                 If ``None``, the input data is not bound to any parameters.
+                If a :class:`~qiskit_machine_learning.circuit.library.QNNCircuit` is passed as
+                circuit, this input is omitted.
```

Comment:
Saying it another way - maybe we can say something more like this.

`If a QNNCircuit is provided the input_params value here is ignored and the value instead is taken from the QNNCircuit input_parameters.` where I used ignored rather than omitted.

Similar for weight_params. And maybe the text for circuit can be updated - something similar is said there